tcl/util_list.tcl

```diff
@@ -103,9 +103,22 @@ proc qc::lunique {list} {
 proc qc::lequal {a b} {
     #| Compare 2 list 
     # Author RS Tcl wiki
-    if {[llength $a] != [llength $b]} {return 0}
-    if {[lindex $a 0] == $a} {return [string equal $a $b]}
-    foreach i $a j $b {if {![lequal $i $j]} {return 0}}
+    if { ! [string is list $a]
```

Comment:
can you add a short comment to each if statement eg. `#a or b not a list`, `a and b not the same length` etc